services/provider/server/server.go

```diff
@@ -363,3 +399,8 @@ func mustMarshal(data map[string]string) []byte {
 	}
 	return newData
 }
+
+func hash(s string) string {
+	h := sha256.Sum256([]byte(s))
+	return hex.EncodeToString(h[:16])
+}
```

Comment:
Why do we need a util function to a 2 liner that is used only once? Can't we just embed this in the call site?